lib/engines/with_migrations.rb

```diff
@@ -0,0 +1,24 @@
+module Engines
+  # It DRYies the common snippet to append engine's migrations to the main app
+  module WithMigrations
```

Comment:
I tried it but AFAIK it won't be easy. We'll have to either force load `shared` before other engines or  add shared as dependency. I tried the latter with no luck.

I can still require the lib from `shared` but, then what is the responsibility of the main app?